### PR TITLE
[RW-7834][risk=no] Require TOS acceptance on login when users are not Terra-TOS compliant

### PR DESCRIPTION
**Pull request by NehaBroad**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 1]
```diff
@@ -466,14 +466,10 @@ public void validateAllOfUsTermsOfService(Integer tosVersion) {
...
   public boolean validateAllOfUsTermsOfServiceVersion(@Nonnull DbUser dbUser) {
-    try {
-      final int tosVersion =
-          userTermsOfServiceDao.findByUserIdOrThrow(dbUser.getUserId()).getTosVersion();
-      return tosVersion == LATEST_AOU_TOS_VERSION;
-    } catch (BadRequestException ex) {
-      // In case user doesnt have any terms of service
-      return false;
-    }
+    return userTermsOfServiceDao
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:469]: woo

**Review by jmthibault79 — CHANGES_REQUESTED** [review]
Please update UserServiceTest with the following:
* update `test_validateTermsOfService_dbUser()` to check for return value
* add a version of this test ^ where `mockFireCloudService.getUserTermsOfServiceStatus()` returns false
* add standalone tests for `validateAllOfUsTermsOfServiceVersion()`

File: api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java [hunk 2]
```diff
@@ -571,40 +571,68 @@ public void test_validateAllOfUsTermsOfService_wrong_version() {
...
+      throws org.pmiops.workbench.firecloud.ApiException {
+    when(mockFireCloudService.getUserTermsOfServiceStatus()).thenReturn(true);
+    DbUser user = createUserWithAoUTOSVersion(LATEST_AOU_TOS_VERSION - 1);
+    assertThat(userService.validateTermsOfService(user)).isFalse();
+  }
+
+  @Test
+  public void test_validateTermsOfService_dbUser_wrong_aou_version_hasNot_acceptedTerra()
+      throws org.pmiops.workbench.firecloud.ApiException {
+    when(mockFireCloudService.getUserTermsOfServiceStatus()).thenReturn(false);
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java:628]: test_validateTermsOfService_dbUser_hasNotAcceptedTerraTOS()  line # 593

[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java:628]: right!  sorry

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**